### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by alkuzich**

### CODE REVIEW

**Review by azjaz — APPROVED** [review]

File: src/test/java/tests/ShoppingCartTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        assertThat(shoppingCart.getProducts()).contains(product);
+    }
+
+    @Test
+    public void testAddProductsToShoppingCartViaConstructor() {
+        List<Product> products = new ArrayList<>();
+        Product product = new Product();
+        products.add(product);
+        ShoppingCart shoppingCart = new ShoppingCart(products);
+        shoppingCart.addProductToCart(product);
```
Comment (Yauhenda) [inline src/test/java/tests/ShoppingCartTest.java]: Why do we need it while we are testing adding via constructor?
After this test we would never know if was it added via the constructor or via the method.

File: src/test/java/tests/ShoppingCartTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import static org.assertj.core.api.Assertions.*;
+
+public class ShoppingCartTest {
+
+    @Test
+    public void testAddProductsToShoppingCart() {
+        ShoppingCart shoppingCart = new ShoppingCart(new ArrayList<>());
+        Product product = new Product();
+        shoppingCart.addProductToCart(product);
+        assertThat(shoppingCart.getProducts()).contains(product);
```
Comment (Yauhenda) [inline src/test/java/tests/ShoppingCartTest.java:20]: I would recommend adding the second check on the size of the cart

**Review by Yauhenda — CHANGES_REQUESTED** [review]

File: src/test/java/tests/ShoppingCartTest.java (hunk 2 above)
Comment (alkuzich) [reply src/test/java/tests/ShoppingCartTest.java:20]: Added check that the shopping card is empty after the only product removal

File: src/test/java/tests/ShoppingCartTest.java (hunk 1 above)
Comment (alkuzich) [reply src/test/java/tests/ShoppingCartTest.java]: Yes, indeed, the line 29 was unnecessary. Fixed.

**Review by Yauhenda — APPROVED** [review]

**Pull request merged**